lib/handler/file.c

```diff
@@ -654,15 +654,35 @@ static int serve_with_generator(struct st_h2o_sendfile_generator_t *generator, h
         }
     }
 
-    /* only allow GET or POST for static files */
+    /* only allow GET or HEAD for static files */
     if (method_type == METHOD_IS_OTHER) {
         do_close(&generator->super, req);
         send_method_not_allowed(req);
         return 0;
     }
 
-    /* if-range */
+    /* range request */
     if ((range_header_index = h2o_find_header(&req->headers, H2O_TOKEN_RANGE, SIZE_MAX)) != -1) {
+        /* if range */
+        if ((if_range_header_index = h2o_find_header(&req->headers, H2O_TOKEN_IF_RANGE, SIZE_MAX)) != -1) {
+            h2o_iovec_t *if_range = &req->headers.entries[if_range_header_index].value;
```

Comment:
Could we extract the strong etag comparison logic as a dedicated function, so that we can have unit tests / call in from other parts of the code?